code/modules/fishing/fishing_gear.dm

```diff
@@ -233,27 +233,31 @@ TYPEINFO(/obj/item/fish_portal)
 		sleep(rand(30,70))
 		// Dispense processed stuff
 		for(var/obj/item/fish/P in src.contents)
-			switch( P.value )
-				if (FISH_RARITY_COMMON)
-					new/obj/item/requisition_token/fishing/common(src.loc, src.layer + 0.1)
-					JOB_XP(user, "Angler", 1)
-					qdel( P )
-				if (FISH_RARITY_UNCOMMON)
-					new/obj/item/requisition_token/fishing/uncommon(src.loc, src.layer + 0.1)
-					JOB_XP(user, "Angler", 2)
-					qdel( P )
-				if (FISH_RARITY_RARE)
-					new/obj/item/requisition_token/fishing/rare(src.loc, src.layer + 0.1)
-					JOB_XP(user, "Angler", 3)
-					qdel( P )
-				if (FISH_RARITY_EPIC)
-					new/obj/item/requisition_token/fishing/epic(src.loc, src.layer + 0.1)
-					JOB_XP(user, "Angler", 4)
-					qdel( P )
-				if (FISH_RARITY_LEGENDARY)
-					new/obj/item/requisition_token/fishing/legendary(src.loc, src.layer + 0.1)
-					JOB_XP(user, "Angler", 5)
-					qdel( P )
+			//No botany fish. Be a real angler and use that damn fishing rod
+			if (P.fishing_upload_blacklisted)
+				P.set_loc(get_turf(src))
```

Comment:
I feel like this should maybe give some message?

Reply:
Or perhaps track if any fish get rejected and display one message at the end if yes. To prevent it from being spammy.

Reply:
That makes certainly sense, will add a little message right now.